### PR TITLE
External PETSc problem

### PR DESCRIPTION
**Pull request by fdkong**
Definitely did not get it done yet, but it is a good starting point for discussion. 

@friedmud

### CODE REVIEW

**Comment by friedmud** [comment]
Let's make sure to talk about this tomorrow...

**Comment by YaqiWang** [comment]
I can see that this is a nice tool for me to learn PETSc. Is there any other purpose for this PR besides educational? Just curious.

**Comment by fdkong** [comment]
> I can see that this is a nice tool for me to learn PETSc. Is there any other purpose for this PR besides educational? Just curious.

It is used to demonstrate how to couple an external APP through MOOSE APP Wrapper (External problem).

File: modules/external_petsc_solver/test/tests/external_petsc_problem/external_petsc_master.i [hunk 1]
```diff
@@ -0,0 +1,91 @@
+###########################################################
+# This is a test of the Multiapp System. This test solves
+# four independent applications spaced throughout a
+# master domain interleaved with a master solve.
+#
+# @Requirement F7.10
+###########################################################
```
Comment (permcody) [inline modules/external_petsc_solver/test/tests/external_petsc_problem/external_petsc_master.i]: Delete this header. We aren't using this @Requirement syntax anymore.

File: modules/external_petsc_solver/test/tests/kernels/simple_diffusion/simple_diffusion.i [hunk 2]
```diff
@@ -0,0 +1,44 @@
+[Mesh]
```
Comment (permcody) [inline modules/external_petsc_solver/test/tests/kernels/simple_diffusion/simple_diffusion.i]: If you have a real test, feel free to delete this simple diffusion test. It's just nice to have a single placeholder.

File: modules/modules.mk [hunk 3]
```diff
@@ -180,6 +181,13 @@ ifeq ($(XFEM),yes)
   include $(FRAMEWORK_DIR)/app.mk
 endif
 
+ifeq ($(EXTERNAL_PETSC_SOLVER),yes)
+  APPLICATION_DIR    := $(MOOSE_DIR)/modules/external_petsc_solver
+  APPLICATION_NAME   := external_petsc_solver
+
```
Comment (permcody) [inline modules/modules.mk]: Oh you need a suffix - that's super important. In fact I'm surprised any of this works without it!

**Review by permcody — CHANGES_REQUESTED** [review]
Getting really close.

**Review by permcody — APPROVED** [review]

**Pull request merged**